worlds/zillion/__init__.py

```diff
@@ -0,0 +1,374 @@
+            if here_name in done:
+                continue
+            here = all[here_name]
+
+            for zz_loc in zz_here.locations:
+                # if local gun reqs didn't place item
+                if not zz_loc.item:
+
+                    def access_rule_wrapped(zz_loc_local: ZzLocation,
+                                            p: int,
+                                            zz_r: ZzRandomizer,
+                                            id_to_zz_item: Dict[int, ZzItem],
+                                            cs: CollectionState) -> bool:
+                        # print(f"checking access to {zz_loc_local}")
+                        accessible = cs_to_zz_locs(cs, p, zz_r, id_to_zz_item)
+                        return zz_loc_local in accessible
+
+                    access_rule = functools.partial(access_rule_wrapped,
+                                                    zz_loc, self.player, self.zz_system.randomizer, self.id_to_zz_item)
+
+                    loc_name = self.zz_system.randomizer.loc_name_2_pretty[zz_loc.name]
+                    loc = ZillionLocation(zz_loc, self.player, loc_name, here)
+                    loc.access_rule = access_rule
+                    if not (limited_skill >= zz_loc.req):
+                        loc.progress_type = LocationProgressType.EXCLUDED
+                        self.world.exclude_locations[p].value.add(loc.name)
+                    here.locations.append(loc)
+                    self.my_locations.append(loc)
+
+            for zz_dest in zz_here.connections.keys():
+                dest_name = "Menu" if zz_dest.name == 'start' else zz_reg_name_to_reg_name(zz_dest.name)
+                dest = all[dest_name]
+                exit = Entrance(p, f"{here_name} to {dest_name}", here)
+                here.exits.append(exit)
+                exit.connect(dest)
+
+                queue.append(zz_dest)
+            done.add(here.name)
+
+    def create_items(self) -> None:
+        if not self.id_to_zz_item:
+            self._make_item_maps("JJ")
+            self.logger.warning("warning: called `create_items` without calling `generate_early` first")
+        assert self.id_to_zz_item, "failed to get item maps"
+
+        # in zilliandomizer, the Randomizer class puts empties in the item pool to fill space,
+        # but here in AP, empties are in the options from options.validate
+        item_counts = self._item_counts
+        self.logger.debug(item_counts)
+
+        for item_name, item_id in self.item_name_to_id.items():
+            zz_item = self.id_to_zz_item[item_id]
+            if item_id >= (4 + base_id):  # normal item
+                if item_name in item_counts:
+                    count = item_counts[item_name]
+                    self.logger.debug(f"Zillion Items: {item_name}  {count}")
+                    for _ in range(count):
+                        self.world.itempool.append(self.create_item(item_name))
```

Comment:
i believe `self.world.itempool += [self.create_item(item_name) for _ in range(count)]` is the nicer and faster way to write this